src/storage/src/table/batch_table/storage_table.rs

```diff
@@ -445,15 +440,34 @@ impl<S: StateStore> StorageTable<S> {
             );
             None
         } else {
-            let pk_prefix_serializer = self.pk_serializer.prefix(pk_prefix.len());
-            let serialized_pk_prefix = serialize_pk(&pk_prefix, &pk_prefix_serializer);
-            Some(serialized_pk_prefix)
+            let encoded_prefix = if let Bound::Included(start_key) = start_key.as_ref() {
+                start_key
+            } else {
+                unreachable!()
```

Comment:
Is this really unreachable? The implementation of `serialize_pk_bound` seems to suggest taht exclusive start key is possible. Based on the comments from L388-389, I think it is also safe to use the encoded exclusive start key here since the pk_prefix should remain unchanged.